### PR TITLE
Bench new univ demote api for elpi

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
not sure this got the right elpi version

**Comment by SkySkimmer** [comment]
This got the wrong elpi version, it took 2.2.0 instead of dev

File: dev/bench/bench.sh [hunk 1]
```diff
@@ -70,15 +70,15 @@ check_variable () {
...
+: "${new_coq_opam_archive_git_branch:=bench-elpi-demote}"
 : "${old_coq_opam_archive_git_branch:=master}"
 : "${new_coq_version:=dev}"
 : "${old_coq_version:=dev}"
 : "${num_of_iterations:=1}"
 : "${timeout:=3h}"
-: "${coq_opam_packages:=coq-test-suite coq-bignums coq-hott coq-performance-tests-lite coq-engine-bench-lite coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-math-classes coq-corn coq-compcert coq-equations coq-metacoq-template coq-metacoq-pcuic coq-metacoq-safechecker coq-metacoq-erasure coq-metacoq-translations coq-color coq-coqprime coq-coqutil coq-bedrock2 coq-rewriter coq-fiat-core coq-fiat-parsers coq-fiat-crypto-with-bedrock coq-unimath coq-coquelicot coq-iris-examples coq-verdi coq-verdi-raft coq-fourcolor coq-rewriter-perf-SuperFast coq-vst coq-category-theory coq-neural-net-interp-computed-lite}"
+: "${coq_opam_packages:=coq-test-suite coq-bignums coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-coquelicot coq-iris-examples coq-fourcolor}"
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:81]:
```suggestion
: "${coq_opam_packages:=coq-elpi.dev coq-bignums coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-coquelicot coq-iris-examples coq-fourcolor}"
```

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by gares** [comment]
@SkySkimmer would it be possible to bench also math comp analysis?

File: dev/bench/bench.sh [hunk 2]
```diff
@@ -70,15 +70,15 @@ check_variable () {
...
+: "${new_coq_opam_archive_git_branch:=bench-elpi-demote}"
 : "${old_coq_opam_archive_git_branch:=master}"
 : "${new_coq_version:=dev}"
 : "${old_coq_version:=dev}"
 : "${num_of_iterations:=1}"
 : "${timeout:=3h}"
-: "${coq_opam_packages:=coq-test-suite coq-bignums coq-hott coq-performance-tests-lite coq-engine-bench-lite coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-math-classes coq-corn coq-compcert coq-equations coq-metacoq-template coq-metacoq-pcuic coq-metacoq-safechecker coq-metacoq-erasure coq-metacoq-translations coq-color coq-coqprime coq-coqutil coq-bedrock2 coq-rewriter coq-fiat-core coq-fiat-parsers coq-fiat-crypto-with-bedrock coq-unimath coq-coquelicot coq-iris-examples coq-verdi coq-verdi-raft coq-fourcolor coq-rewriter-perf-SuperFast coq-vst coq-category-theory coq-neural-net-interp-computed-lite}"
+: "${coq_opam_packages:=coq-elpi.dev coq-bignums coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-coquelicot coq-iris-examples coq-fourcolor}"
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:81]:
```suggestion
: "${coq_opam_packages:=coq-elpi.dev coq-bignums coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-mathcomp-finmap coq-mathcomp-classical coq-mathcomp-bigenough coq-mathcomp-analysis}"
```

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by gares** [comment]
I looked at the html files. The pattern in ssrnum is the expected one, since each call to HB.structure is a bit faster.
In order, on the contrary, many calls are slower but there a couple of calls to HB.saturate are faster.
Let's see if they are reproducible.

**Pull request closed without merging**